### PR TITLE
Stop evaluation of conditions when one evaluates to true

### LINKED ISSUE #3102: [Conditionals] Short circuit condition evaluation for performance
Currently [we exhaustively evaluate every condition each time new data is received](https://github.com/nasa/openmct/blob/771fb9c04464480e3bacc6ed482b622b3f6fafeb/src/plugins/condition/ConditionManager.js#L314). We could short circuit this by stopping on the first condition that evaluates to `true`. Perhaps it would make sense to use [Array.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) for this purpose?

### PR DESCRIPTION
**Pull request by shefalijoshi**
Resolves #3102

### Author Checklist
1. Changes address original issue? Yes
2. Unit tests included and/or updated with changes? Yes
3. Command line build passes? Yes
4. Changes have been smoke-tested? Yes
5. Testing instructions included? Yes

### CODE REVIEW

File: src/plugins/condition/ConditionManager.js [hunk 1]
```diff
@@ -322,8 +322,11 @@ export default class ConditionManager extends EventEmitter {
         let timestamp = {};
         timestamp[timeSystemKey] = normalizedDatum[timeSystemKey];
 
-        this.conditions.forEach(condition => {
+        //We want to stop when the first condition evaluates to true.
+        this.conditions.some((condition) => {
```
Comment (nikhilmandlik) [inline src/plugins/condition/ConditionManager.js:326]: you can return `result` from `getResult` in `openmct/src/plugins/condition/Condition.js`, then 
`this.conditions.some(condition =>  condition.getResult(normalizedDatum));`

Comment (shefalijoshi) [reply src/plugins/condition/ConditionManager.js:326]: Yeah, it's sort of the same thing. I think this is less of a change to the files. I'd prefer to leave it this way.

Comment (nikhilmandlik) [reply src/plugins/condition/ConditionManager.js:326]: hmm, found it confusing because of name `getResult` and its not returning any result.

**Comment by davetsay** [comment]
What happens if a condition changes? for instance if telemetry is removed. Do the conditions get recalculated in this case? If so, then this looks good to me.

**Comment by shefalijoshi** [comment]
> What happens if a condition changes? for instance if telemetry is removed. Do the conditions get recalculated in this case? If so, then this looks good to me.

They would get recalculated when the next telemetry datum arrives - same as before.

File: src/plugins/condition/ConditionManager.js (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/condition/ConditionManager.js:328]: I've changed the method name to updateResult in Condition and Criterion classes.

**Comment by nikhilmandlik** [comment]
### Reviewer Checklist
Check | Passed
-- | --
Changes appear to address issue? | Y
Appropriate unit tests included? | NA
Code style and in-line documentation are appropriate? | Y
Commit messages meet standards? | Y
Has associated issue been labelled unverified? (only applicable if this PR closes the issue) | Y

**Pull request merged**